scripts/lint.js

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env node
+
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { execSync } from 'node:child_process';
+import { mkdirSync, rmSync } from 'node:fs';
+import { tmpdir } from 'node:os';
+import { join } from 'node:path';
+
+const ACTIONLINT_VERSION = '1.7.7';
+const SHELLCHECK_VERSION = '0.11.0';
+const YAMLLINT_VERSION = '1.35.1';
+
+const TEMP_DIR = join(tmpdir(), 'gemini-cli-linters');
+
+function getPlatformArch() {
+  const platform = process.platform;
+  const arch = process.arch;
+  if (platform === 'linux' && arch === 'x64') {
+    return {
+      actionlint: 'linux_amd64',
+      shellcheck: 'linux.x86_64',
+    };
+  }
+  if (platform === 'darwin' && arch === 'x64') {
+    return {
+      actionlint: 'darwin_amd64',
+      shellcheck: 'darwin.x86_64',
+    };
+  }
+  if (platform === 'darwin' && arch === 'arm64') {
+    return {
+      actionlint: 'darwin_arm64',
+      shellcheck: 'darwin.aarch64',
+    };
+  }
+  throw new Error(`Unsupported platform/architecture: ${platform}/${arch}`);
+}
+
+const platformArch = getPlatformArch();
+
+const LINTERS = {
```

Comment:
Consider one of the following:

- Give this object a structured type

```
interface Linter {
  check: string;
  installer: string;
  run: string;
}

interface Linters {
  actionlint: Linter;
  whateverlint: Linter;
  etcetc: Linter;
}
```

OR

- define this object with `as const`

`const LINTER = {....} as const;`

OR 

(Best option)

Do both

`const LINTER: Linters  = {...} as const satisfies Linters;`